### PR TITLE
Homepage now redirects to a new search page for the centralized search + random episode button

### PR DESCRIPTION
**Pull request by batoulyehia**
In this pull request, I fixed the redirect to home page when the app is first launched, and it leads to another page when you press on the search button on top.

### CODE REVIEW

**Comment by batoulyehia** [comment]
UI tests have now been modified to be consistent with new design

File: app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java [hunk 1]
```diff
@@ -30,11 +38,48 @@ public void onCreate(Bundle savedInstanceState) {
...
+        inflater.inflate(R.menu.home_toolbar, menu);
+    }
+
+    //Handles the options selected
+    //Redirects to the search page with the random episode button
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item){
+        switch(item.getItemId()){
+            case R.id.search_home:
+                Log.d("Home Search","Redirecting to centralized search results...");
```
Comment (SaiShan) [inline app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java]: Log.d

File: app/src/main/java/de/danoeh/antennapod/fragment/PodSearchFragment.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        View root = inflater.inflate(R.layout.centralizedsearch_results, container, false);
+        if (getActivity() instanceof MainActivity) {
+            ((MainActivity) getActivity()).getSupportActionBar().setTitle("Search for a Podcast");
+        }
+
```
Comment (OlivierTD) [inline app/src/main/java/de/danoeh/antennapod/fragment/PodSearchFragment.java]: Extra space

File: app/src/main/res/layout/centralizedsearch_results.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        android:gravity="center"
+        android:layout_marginTop="10dp"/>
+
+    <fragment
+        android:layout_marginTop="10dp"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:name="de.danoeh.antennapod.fragment.RandomPodcastFragment"
+        android:id="@+id/randomEpisodeFrag"
+
```
Comment (OlivierTD) [inline app/src/main/res/layout/centralizedsearch_results.xml]: Extra space

File: app/src/main/res/layout/centralizedsearch_results.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/searchScrollView"
+    android:layout_width="fill_parent"
+    android:layout_height="fill_parent">
+    <LinearLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        android:orientation="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
```
Comment (OlivierTD) [inline app/src/main/res/layout/centralizedsearch_results.xml]: Extra space

File: app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java [hunk 5]
```diff
@@ -30,11 +38,48 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         super.onCreateView(inflater, container, savedInstanceState);
-        View root = inflater.inflate(R.layout.home_page, container, false);
-
+        if (getActivity() instanceof MainActivity) {
```
Comment (OlivierTD) [inline app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java]: Any reason why you have an if statement here? I don't understand its use.

File: app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java [hunk 6]
```diff
@@ -30,11 +38,48 @@ public void onCreate(Bundle savedInstanceState) {
...
+            ((MainActivity) getActivity()).getSupportActionBar().setTitle("Home");
+        }
+        try {
```
Comment (OlivierTD) [inline app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java]: Do you really need a try-catch here? Explain to me if you do.

File: app/src/main/java/de/danoeh/antennapod/fragment/PodSearchFragment.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        View root = inflater.inflate(R.layout.centralizedsearch_results, container, false);
+        if (getActivity() instanceof MainActivity) {
```
Comment (OlivierTD) [inline app/src/main/java/de/danoeh/antennapod/fragment/PodSearchFragment.java]: Same here why do you need an if statement? Can you explain to me?

**Review by OlivierTD — CHANGES_REQUESTED** [review]
I will separate my review into three parts: the functionality of the app with this PR, the code, and the commits.
1. The functionality of the app:
There are few things that are bugging me about how the centralized search function. I thought that we were going to add a random button on the title bar to the right of the search button that would automatically pop a random search. I guess that we can keep it the way it is if most people actually prefer it the way you did it. The other Thing that bugs me, when I click on the search button, I have to click again on it to have the keyboard pop and have me input my keyword for the search. This is not intuitive. I think that a typical user will expect to enter right away keywords when they click on that search button. Now, they have to press it twice. Let me know what you think of the two things that are bugging me, and what you will do to solve them. Moreover, when I conduct a search, there is a space between the random button and the list generated. Can the space be smaller? It looks a bit weird otherwise. Also, when I open the app, it does not bring me the home page, it brings me to the page that I left off, I am not sure if it is what we intended to do. I thought that whenever we open the app, we wanted it to land on the home page. Lastly, I do not think that we should display to the user the category of the random search, I think that it should just give them right away a random episode, no matter the category!
2. The code:
I left couple comments throughout your code. Most of it is fine and I am being very picky about extra spaces. Try to have your code as neat as possible since it facilitates reading and reviewing it.
3. The commits:
Good job referencing your issues in your commits. Some of your commit messages are too long. There are two parts to commits: the title and the message. If you can fit in the title a short description of what the commit does, it doesn't need a message body. If the title is going to be too long with the description, you should be using a message along with the title. You should look online to figure out how to have a title and a message if you don't know how to do it. Some commits are too short or useless so try to be careful before commiting. In one commit, you add code that you remove in the commit after. [Here](https://github.com/OlivierTD/team-rocket/pull/153/commits/5a0bd75e7e8ce8e4632acbc7989a999083975d9a) and [here](https://github.com/OlivierTD/team-rocket/pull/153/commits/8fb3c8e26de3eaf7f3f2c75a5518da2661674c76). If you change only one line, try to include it into another more significant commit. [Here](https://github.com/OlivierTD/team-rocket/pull/153/commits/03fcf745bb133c2e8ede0a3c10afa70d299e1ab1) you changed one line and I think that it is still not working for me? Before commiting something, always verify that the code actually works 100%!

If you have any questions or comments let me know, thanks Batoul!

File: app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java (hunk 6 above)
Comment (batoulyehia) [reply app/src/main/java/de/danoeh/antennapod/fragment/HomePageFragment.java]: Yes, that was the solution to fix the back button functionality. Without it, I kept getting an error saying that the class couldn't be inflated.

**Comment by batoulyehia** [comment]
I know that the idea was to have a button on the action bar for the random podcast button, however, when I tried implementing it, the application would crash because I removed the centralized search fragment from the homepage fragment. It only worked when both the centralized fragment and the random episode button were on the same page. I'm looking into modifying it though and try to get it work the other way.
Thank you for review Olivier, I'll do my best to make it work.

**Comment by OlivierTD** [comment]
I do not see how both these buttons should be clashing. Try to talk to whoever implemented the code if you can't figure it out.

**Comment by batoulyehia** [comment]
The functionality now works!!!!! 😄 
I made it so that when you get redirected to the next page, the keyboard appears and the view of the action bar is expanded. So, the user doesn't have to click twice on the search button, it won't bug them.

**Review by OlivierTD — CHANGES_REQUESTED** [review]
there is still an issue with this feature. When I press the search I get a keyboard that pops out. If I press the back button, it removes the keyboard and keeps me on the search for podcast page. Also, there is no search button on the search for a podcast page. I only see a three-dot button that is acting like the search. This is could be confusing to the users.

I don't understand why when we press the search button on the home page it brings us to a new fragment. I thought that we wanted to generate a new fragment when the search action is sent. I feel like the random button part takes too much space on the search fragment. This is why I thought we were gonna keep a small button on the home page for the random search.

**Comment by batoulyehia** [comment]
The issue with keeping the random button is that if the random button stays on the home page, it won't work. The application will crash as long as the search results aren't loaded on the homepage itself. That was my solution to it. Nothing else worked. And if you press on the back button, the keyboard will disappear, but if you press it again then you'll be redirected to the homepage.

**Comment by batoulyehia** [comment]
Functionality working now like how we discussed it before. Ready for merge.

**Review by OlivierTD — CHANGES_REQUESTED** [review]
The random button does not have the right color under the dark theme.

When I do another search while on the result page and I press back, I am shown an empty page. It would be better if a user would be redirected right away to the home page.

Also when you do a search it would be great if the title would be Search Result or something more suitable than "Home".

**Comment by batoulyehia** [comment]
I did some refactoring, and now what it does when you press back is that it returns you to the previous search you made. I'll fix the title right away though

**Comment by OlivierTD** [comment]
Tests are failing on your branch and codeclimate has issues for you to address.

**Comment by OlivierTD** [comment]
You have conflicts with the master branch now.

**Pull request closed without merging**